### PR TITLE
Add unittest for voting_utils and fix a bug.

### PR DESCRIPTION
**Pull request by wen-coding**
- Add unittest for voting_utils
- Fix a bug found in unittest: Right now we use root bank's epoch to get bls_pubkey_to_rank_map, we should really use the epoch of the slot being voted on.

### CODE REVIEW

File: votor/src/voting_utils.rs [hunk 1]
```diff
@@ -313,3 +311,293 @@ pub fn generate_vote_message(
...
+            own_vote_sender,
+            bls_sender: unbounded().0,
+            commitment_sender: unbounded().0,
+            wait_to_vote_slot: None,
+            root_bank,
+        }
+    }
+
+    #[test]
+    fn test_generate_vote_message() {
```
Comment (carllin) [inline votor/src/voting_utils.rs]: nit: test_generate_own_vote_message()

File: votor/src/voting_utils.rs [hunk 2]
```diff
@@ -313,3 +311,293 @@ pub fn generate_vote_message(
...
+            panic!("Expected BLSOp::PushVote, got {result:?}");
+        }
+
+        // Check that own vote sender receives the vote
+        let received_message = own_vote_receiver.recv().unwrap();
+        assert_eq!(received_message, expected_message);
+    }
+
+    #[test]
+    fn test_wrong_vote_account() {
```
Comment (carllin) [inline votor/src/voting_utils.rs]: nit: this covers more than just the wrong vote account. Maybe  break the other edge cases into separate tests

File: votor/src/voting_utils.rs [hunk 3]
```diff
@@ -313,3 +311,293 @@ pub fn generate_vote_message(
...
+        assert!(generate_vote_message(vote, false, &mut voting_context)
+            .unwrap()
+            .is_some());
+
+        // Wrong identity keypair
+        voting_context.identity_keypair = Arc::new(Keypair::new());
+        let vote = Vote::new_notarization_vote(6, Hash::new_unique());
+        assert!(generate_vote_message(vote, true, &mut voting_context)
+            .unwrap()
+            .is_none());
```
Comment (carllin) [inline votor/src/voting_utils.rs:516]: Do these types of mismatched keys not return VoteError's? Do we have tests to cover the different types of VoteErrors?

File: votor/src/voting_utils.rs [hunk 4]
```diff
@@ -313,3 +311,293 @@ pub fn generate_vote_message(
...
+        let mut voting_context =
+            setup_voting_context_and_bank_forks(own_vote_sender, &validator_keypairs, my_index);
+
+        // If we try to vote for a slot in the future, we should panic
+        let vote = Vote::new_notarization_vote(1_000_000_000, Hash::new_unique());
+        let _ = generate_vote_message(vote, false, &mut voting_context);
+    }
+
+    #[test]
+    fn test_zero_staked_validator() {
```
Comment (carllin) [inline votor/src/voting_utils.rs]: nit: test_zero_staked_validator_fails_voting

File: votor/src/voting_utils.rs (hunk 1 above)
Comment (wen-coding) [reply votor/src/voting_utils.rs]: Done.

File: votor/src/voting_utils.rs (hunk 2 above)
Comment (wen-coding) [reply votor/src/voting_utils.rs]: Done.

File: votor/src/voting_utils.rs (hunk 3 above)
Comment (wen-coding) [reply votor/src/voting_utils.rs:516]: Our theory is that these can all happen on a hot spare when validator operators are busy configuring their hot spare validators correctly. So right now it's all INFO logs, maybe we should upgrade the logs to warning/error instead?

File: votor/src/voting_utils.rs (hunk 4 above)
Comment (wen-coding) [reply votor/src/voting_utils.rs]: Done.

File: votor/src/voting_utils.rs (hunk 3 above)
Comment (carllin) [reply votor/src/voting_utils.rs:516]: yeah that makes sense to me as a warning

**Review by carllin — APPROVED** [review]
